src/reanimated2/hook/useScrollViewOffset.ts

```diff
@@ -100,11 +101,15 @@ function useScrollViewOffsetNative(
   ) as unknown as EventHandlerInternal<ReanimatedScrollEvent>;
 
   useEffect(() => {
-    const component = animatedRef.current;
-    const viewTag = IS_WEB ? component : findNodeHandle(component);
+    // We need to make sure that listener for old animatedRef value is removed
+    if (scrollRef.current !== null) {
+      eventHandler.workletEventHandler?.unregisterFromEvents();
```

Comment:
Is it necessary here? Why not just
```suggestion
      eventHandler.workletEventHandler.unregisterFromEvents();
```